packages/control-hub/src/internal/ControlHub.ts

```diff
@@ -0,0 +1,649 @@
+
+                if (callback) {
+                    callback(response, error);
+                }
+            }
+        });
+
+        this.webSocketConnection.on("close", () => {
+            console.log(`Connection was closed`);
+            this.isConnected = false;
+        });
+
+        this.webSocketConnection.on("error", (_: WebSocket, err: Error) => {
+            console.log("Websocket error");
+            console.log(err);
+            this.isConnected = false;
+        });
+
+        return new Promise((resolve, reject) => {
+            this.webSocketConnection.on("open", async () => {
+                this.isConnected = true;
+                await this.subscribe();
+                let apiVersion: { majorVersion: number; minorVersion: number } =
+                    await this.sendCommand("start", {});
+
+                this.id = await this.openHub("(embedded)", this.moduleAddress);
+
+                this.embedded = new ControlHubConnectedExpansionHub(
+                    true,
+                    RevHubType.ControlHub,
+                    this.sendCommand.bind(this),
+                    "(embedded)",
+                    this.moduleAddress,
+                    this.id,
+                );
+
+                if (
+                    apiVersion.majorVersion !== this.supportedManualControlMajorVersion ||
+                    apiVersion.minorVersion < this.supportedManualControlMinorVersion
+                ) {
+                    reject(
+                        new Error(
+                            `API Version ${apiVersion.majorVersion}.${apiVersion.minorVersion} is not supported`,
+                        ),
+                    );
+                }
+
+                resolve();
+            });
+        });
+    }
+
+    async isWiFiConnected(): Promise<boolean> {
+        try {
+            let response = await axios.get("http://192.168.43.1:8080/js/rcInfo.json", {
+                timeout: 1000,
+            });
+            if (response.data) {
+                let rcVersion: string = response.data.rcVersion;
+                return semver.satisfies(rcVersion, ">=8.2");
+            }
+
+            return false;
+        } catch (e) {
+            return false;
+        }
+    }
+
+    async subscribe(): Promise<void> {
+        let payload = {
+            namespace: "system",
+            type: "subscribeToNamespace",
+            payload: "MC",
+        };
+        this.webSocketConnection.send(JSON.stringify(payload));
+    }
+
+    async openHub(
+        serialNumber: string,
+        parentAddress: number,
+        address: number = parentAddress,
+    ): Promise<any> {
+        return await this.sendCommand("openHub", {
+            parentSerialNumber: serialNumber,
+            parentHubAddress: parentAddress,
+            hubAddress: address,
+        });
+    }
+
+    close() {
+        this.embedded.close();
+        this.sendCommand("stop", {}).then(() => this.webSocketConnection.close());
```

Comment:
Should we loop through a flatmap of children here and send a stop command to each?